### DISCUSSION TITLE
What is the benifit of gitlab-ci-local over gitlab-runner exec?

### DISCUSSION THREAD

**Discussion by Vad1mo** [firecow/gitlab-ci-local#755] [Q&A] [answered]
So far, I was using gitlab-runner so far, Now that I discovered [gitlab-ci-local](https://github.com/firecow/gitlab-ci-local) I wonder what benifits and problems it solves over  gitlab-runner?

```
gitlab-runner exec docker my-job-name

```

**Comment by firecow** [comment, marked as answer]
`gitlab-runner exec` is not capable of parsing `extends`, merging `includes` or expanding `!reference [.hidden, script]`, the moment you start having "complex" ci yaml, `gitlab-runner exec` isn't of much help.

`gitlab-ci-local` makes it possible to pass custom variables to your job, making it easier to mimic/modify gitlab's ci/cd variables.

`gitlab-ci-local` will copy artifacts to your local checkout, so that your IDE can index stuff like `node_modules` or php `vendor` folders to provide code autocompletion n' other help.

`gitlab-ci-local` can execute an entire pipeline or a subset of jobs (`--needs`, `--only-needs`), whereas `gitlab-runner exec` only runs a single job.

Also, I'm pretty sure gitlab said `gitlab-runner exec` will be phased out.
